packages/components/src/__alpha__/SingleSelect/subcomponents/Popover/Popover.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, type PropsWithChildren } from 'react'
+import { useOverlayPosition } from '@react-aria/overlays'
+import { FocusScope as RACFocusScope } from 'react-aria'
+import { Popover as RACPopover } from 'react-aria-components'
+import { useSingleSelectContext } from '../../context'
+import styles from './Popover.module.css'
+
+type PopoverProps = {
+  buttonRef: React.RefObject<HTMLElement>
+  popoverRef: React.RefObject<HTMLDivElement>
+  racPopoverRef: React.Ref<any>
+}
+
+export const Popover = ({
+  buttonRef,
+  popoverRef,
+  racPopoverRef,
+  children,
+}: PopoverProps & PropsWithChildren): React.ReactElement => {
+  const { isOpen, setOpen } = useSingleSelectContext()
+
+  useEffect(() => {
+    if (!popoverRef.current?.showPopover || !popoverRef.current?.hidePopover) return
+    if (isOpen) {
+      popoverRef.current.showPopover()
+    } else {
+      popoverRef.current.hidePopover()
+    }
+  }, [isOpen, popoverRef])
+
+  const { overlayProps } = useOverlayPosition({
+    targetRef: buttonRef,
+    overlayRef: popoverRef,
+    placement: 'bottom start',
+    offset: 0,
+    isOpen: isOpen,
+    shouldFlip: true,
+  })
+  return (
+    <RACPopover trigger="manual" isOpen={isOpen} onOpenChange={setOpen} ref={racPopoverRef}>
+      <div
+        // @ts-expect-error - popover attribute is not included in current ts version, ignore type error
+        popover="manual"
+        ref={popoverRef}
+        style={{
+          ...overlayProps,
+          zIndex: 'none',
+          margin: '0',
+          boxSizing: 'border-box',
+          width: buttonRef.current?.getBoundingClientRect().width ?? '200px',
```

Comment:
So this would mean whatever the trigger button size is there popover box will be. Is there plans to give the consumers the ability to control the trigger styles a little more, ie: either a custom trigger button or giving them a className prop?